pkg/pgclient/client.go

```diff
@@ -77,19 +77,28 @@ func getPgConfig(cfg *Config) (*pgxpool.Config, int, error) {
 		return nil, numCopiers, err
 	}
 
-	var pgConfig *pgxpool.Config
+	var (
+		pgConfig          *pgxpool.Config
+		connectionArgsFmt string
+	)
 	if cfg.DbUri == defaultDBUri {
-		pgConfig, err = pgxpool.ParseConfig(connectionStr + fmt.Sprintf(" pool_max_conns=%d pool_min_conns=%d", maxConnections, minConnections))
+		connectionArgsFmt = "%s pool_max_conns=%d pool_min_conns=%d statement_cache_capacity=%d"
 	} else {
-		pgConfig, err = pgxpool.ParseConfig(connectionStr + fmt.Sprintf("&pool_max_conns=%d&pool_min_conns=%d", maxConnections, minConnections))
+		connectionArgsFmt = "%s&pool_max_conns=%d&pool_min_conns=%d&statement_cache_capacity=%d"
 	}
-
+	connectionStringWithArgs := fmt.Sprintf(connectionArgsFmt, connectionStr, maxConnections, minConnections, cfg.MetricsCacheSize)
```

Comment:
Didn't see this change implemented?
```suggestion
	connectionStringWithArgs := fmt.Sprintf(connectionArgsFmt, connectionStr, maxConnections, minConnections, cfg.MetricsCacheSize*2)
```